bundles/org.openhab.binding.vizio/src/main/java/org/openhab/binding/vizio/internal/handler/VizioHandler.java

```diff
@@ -0,0 +1,572 @@
+                    if (debounce && volumeItem.isPresent()) {
+                        currentVolumeHash = volumeItem.get().getHashval();
+
+                        try {
+                            int polledVolume = Integer.parseInt(volumeItem.get().getValue());
+                            if (polledVolume != currentVolume) {
+                                currentVolume = polledVolume;
+                                updateState(VOLUME, new PercentType(BigDecimal.valueOf(currentVolume)));
+                            }
+                        } catch (NumberFormatException e) {
+                            logger.debug("Unable to parse volume value {} as int", volumeItem.get().getValue());
+                        }
+                    }
+
+                    Optional<ItemAudio> muteItem = audioSettings.getItems().stream()
+                            .filter(i -> MUTE.equals(i.getCname())).findFirst();
+                    if (debounce && muteItem.isPresent()) {
+                        String polledMute = muteItem.get().getValue().toUpperCase(Locale.ENGLISH);
+
+                        if (ON.equals(polledMute) || OFF.equals(polledMute)) {
+                            if (ON.equals(polledMute) && !currentMute) {
+                                updateState(MUTE, OnOffType.ON);
+                                currentMute = true;
+                            } else if (OFF.equals(polledMute) && currentMute) {
+                                updateState(MUTE, OnOffType.OFF);
+                                currentMute = false;
+                            }
+                        } else {
+                            logger.debug("Unknown mute mode {}, for response object: {}", polledMute, audioSettings);
+                        }
+                    }
+                } catch (VizioException e) {
+                    logger.debug("Unable to retrieve Vizio TV current audio settings. Exception: {}", e.getMessage(),
+                            e);
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+                }
+            }
+
+            if (powerOn && isLinked(SOURCE)) {
+                try {
+                    CurrentInput polledInputState = communicator.getCurrentInput();
+
+                    if (debounce && !polledInputState.getItems().isEmpty()
+                            && !currentInput.equals(polledInputState.getItems().get(0).getValue())) {
+                        currentInput = polledInputState.getItems().get(0).getValue();
+                        currentInputHash = polledInputState.getItems().get(0).getHashval();
+                        updateState(SOURCE, new StringType(currentInput));
+                    }
+                } catch (VizioException e) {
+                    logger.debug("Unable to retrieve Vizio TV current input. Exception: {}", e.getMessage(), e);
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
```

Comment:
Consider if the message above or even the exception message should be included in this status update to provide more immediate information to the user.